### ISSUE TITLE
Groovy Objects

### ISSUE THREAD

**Issue by hythloda** [deephaven/deephaven-core#1960]
**Description**

version 0.9.0 broke the examples for groovy objects.

This means that the nightly test of snapshots is broken unless we skip these tests. 

**Steps to reproduce**
open a groovy IDE for version 0.9.0
enter below into console
```
class MyObj {
    public int a, b, c

    MyObj(int a, int b, int c) {
        this.a = a
        this.b = b
        this.c = c
    }

    int compute(int value1){
        return  value1
    }
}

obj = new MyObj(1, 2, 3)

source = newTable(
        stringCol("X", "A", "B", "C", "D", "E", "F", "G"),
        intCol("Y", 1, 2, 3, 4, 5, 6, 7),
        intCol("Z", 2, 3, 1, 2, 3, 1, 2)
)

result = source.update("A = obj.a", "B = obj.compute(Y)")
```

**Expected results**
two tables show in IDE
<img width="525" alt="Screen Shot 2022-02-09 at 1 00 39 PM" src="https://user-images.githubusercontent.com/51727488/153262003-50ba5470-c7db-44a7-b24c-070754cc0075.png">

**Actual results**

Error
```
r-Scheduler-Serial-1 | i.d.s.s.SessionState      | Internal Error '86b0b4e6-e211-42dc-9f9c-064715bd341e' io.deephaven.engine.table.impl.select.FormulaCompilationException: Formula compilation error for: obj.a
        at io.deephaven.engine.table.impl.select.DhFormulaColumn.initDef(DhFormulaColumn.java:210)
        at io.deephaven.engine.table.impl.select.SwitchColumn.initDef(SwitchColumn.java:67)
        at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:69)
        at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:41)
        at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$36(QueryTable.java:1377)
        at io.deephaven.engine.table.impl.perf.QueryPerformanceRecorder.withNugget(QueryPerformanceRecorder.java:519)
        at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$37(QueryTable.java:1362)
        at io.deephaven.engine.table.impl.QueryTable.memoizeResult(QueryTable.java:3152)
        at io.deephaven.engine.table.impl.QueryTable.selectOrUpdate(QueryTable.java:1361)
        at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:1341)
        at io.deephaven.engine.table.impl.TableWithDefaults.update(TableWithDefaults.java:287)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:566)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoCachedMethodSite.invoke(PojoMetaMethodSite.java:191)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
        at io.deephaven.dynamic.Script_18.run(Script_18.groovy:34)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:427)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:461)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:436)
        at io.deephaven.engine.util.GroovyDeephavenSession.evaluateCommand(GroovyDeephavenSession.java:229)
        at io.deephaven.engine.util.GroovyDeephavenSession.lambda$evaluate$2(GroovyDeephavenSession.java:252)
        at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:46)
        at io.deephaven.engine.util.GroovyDeephavenSession.evaluate(GroovyDeephavenSession.java:252)
        at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:147)
        at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:71)
        at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:85)
        at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$8(ConsoleServiceGrpcImpl.java:219)
        at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1299)
        at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:847)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.deephaven.server.runner.DeephavenApiServerModule$ThreadFactory.lambda$newThread$0(DeephavenApiServerModule.java:143)
        at java.lang.Thread.run(Thread.java:829)
caused by:
io.deephaven.engine.table.impl.select.FormulaCompilationException: Formula compilation error for: obj.a
        at io.deephaven.engine.table.impl.select.codegen.FormulaAnalyzer.analyze(FormulaAnalyzer.java:33)
        at io.deephaven.engine.table.impl.select.DhFormulaColumn.initDef(DhFormulaColumn.java:187)
        at io.deephaven.engine.table.impl.select.SwitchColumn.initDef(SwitchColumn.java:67)
        at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:69)
        at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:41)
        at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$36(QueryTable.java:1377)
        at io.deephaven.engine.table.impl.perf.QueryPerformanceRecorder.withNugget(QueryPerformanceRecorder.java:519)
        at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$37(QueryTable.java:1362)
        at io.deephaven.engine.table.impl.QueryTable.memoizeResult(QueryTable.java:3152)
        at io.deephaven.engine.table.impl.QueryTable.selectOrUpdate(QueryTable.java:1361)
        at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:1341)
        at io.deephaven.engine.table.impl.TableWithDefaults.update(TableWithDefaults.java:287)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:566)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoCachedMethodSite.invoke(PojoMetaMethodSite.java:191)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
        at io.deephaven.dynamic.Script_18.run(Script_18.groovy:34)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:427)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:461)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:436)
        at io.deephaven.engine.util.GroovyDeephavenSession.evaluateCommand(GroovyDeephavenSession.java:229)
        at io.deephaven.engine.util.GroovyDeephavenSession.lambda$evaluate$2(GroovyDeephavenSession.java:252)
        at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:46)
        at io.deephaven.engine.util.GroovyDeephavenSession.evaluate(GroovyDeephavenSession.java:252)
        at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:147)
        at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:71)
        at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:85)
        at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$8(ConsoleServiceGrpcImpl.java:219)
        at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1299)
        at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:847)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.deephaven.server.runner.DeephavenApiServerModule$ThreadFactory.lambda$newThread$0(DeephavenApiServerModule.java:143)
        at java.lang.Thread.run(Thread.java:829)
caused by:
io.deephaven.engine.table.impl.lang.QueryLanguageParser$QueryLanguageParseException:
Having trouble with the following expression:
Full expression           : obj.a
Expression having trouble : io.deephaven.engine.table.impl.lang.QueryLanguageParser$VisitArgs@6d1d9819
Exception message         : Cannot resolve field name.
    Expression : obj.a
    Scope      : obj
    Scope Type : groovy.lang.GroovyObject
    Field Name : a
        at io.deephaven.engine.table.impl.lang.QueryLanguageParser.visit(QueryLanguageParser.java:1493)
        at io.deephaven.engine.table.impl.lang.QueryLanguageParser.visit(QueryLanguageParser.java:41)
        at com.github.javaparser.ast.expr.FieldAccessExpr.accept(FieldAccessExpr.java:90)
        at io.deephaven.engine.table.impl.lang.QueryLanguageParser.<init>(QueryLanguageParser.java:132)
        at io.deephaven.engine.table.impl.lang.QueryLanguageParser.<init>(QueryLanguageParser.java:85)
        at io.deephaven.engine.table.impl.select.codegen.FormulaAnalyzer.getCompiledFormula(FormulaAnalyzer.java:141)
        at io.deephaven.engine.table.impl.select.codegen.FormulaAnalyzer.analyzeHelper(FormulaAnalyzer.java:47)
        at io.deephaven.engine.table.impl.select.codegen.FormulaAnalyzer.analyze(FormulaAnalyzer.java:31)
        at io.deephaven.engine.table.impl.select.DhFormulaColumn.initDef(DhFormulaColumn.java:187)
        at io.deephaven.engine.table.impl.select.SwitchColumn.initDef(SwitchColumn.java:67)
        at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:69)
        at io.deephaven.engine.table.impl.select.analyzers.SelectAndViewAnalyzer.create(SelectAndViewAnalyzer.java:41)
        at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$36(QueryTable.java:1377)
        at io.deephaven.engine.table.impl.perf.QueryPerformanceRecorder.withNugget(QueryPerformanceRecorder.java:519)
        at io.deephaven.engine.table.impl.QueryTable.lambda$selectOrUpdate$37(QueryTable.java:1362)
        at io.deephaven.engine.table.impl.QueryTable.memoizeResult(QueryTable.java:3152)
        at io.deephaven.engine.table.impl.QueryTable.selectOrUpdate(QueryTable.java:1361)
        at io.deephaven.engine.table.impl.QueryTable.update(QueryTable.java:1341)
        at io.deephaven.engine.table.impl.TableWithDefaults.update(TableWithDefaults.java:287)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:566)
        at org.codehaus.groovy.runtime.callsite.PlainObjectMetaMethodSite.doInvoke(PlainObjectMetaMethodSite.java:43)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoCachedMethodSite.invoke(PojoMetaMethodSite.java:191)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:56)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
        at io.deephaven.dynamic.Script_18.run(Script_18.groovy:34)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:427)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:461)
        at groovy.lang.GroovyShell.evaluate(GroovyShell.java:436)
        at io.deephaven.engine.util.GroovyDeephavenSession.evaluateCommand(GroovyDeephavenSession.java:229)
        at io.deephaven.engine.util.GroovyDeephavenSession.lambda$evaluate$2(GroovyDeephavenSession.java:252)
        at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:46)
        at io.deephaven.engine.util.GroovyDeephavenSession.evaluate(GroovyDeephavenSession.java:252)
        at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:147)
        at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:71)
        at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:85)
        at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$8(ConsoleServiceGrpcImpl.java:219)
        at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1299)
        at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:847)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.deephaven.server.runner.DeephavenApiServerModule$ThreadFactory.lambda$newThread$0(DeephavenApiServerModule.java:143)
        at java.lang.Thread.run(Thread.java:829)
```

**Additional details and attachments**

The correct results are generated in v0.8.0

**Versions**
 - Deephaven: 0.9.0

**Comment by devinrsmith** [comment]
Bug introduced as part of #1522

**Comment by devinrsmith** [comment]
Here's a more target reproduction:

```groovy
class MyObj {
    public int a;
    MyObj(int a) {
        this.a = a
    }
}
obj = new MyObj(1)
result = emptyTable(1).select("A = obj.a")
```

I'll create a quick test to demonstrate that it is currently failing.

**Comment by nbauernfeind** [comment]
This is closed by #2082.
